framework/Samplers/Umbrella.py

```diff
@@ -0,0 +1,311 @@
+    samplerInitInput.addSub(weightTargetInput)
+    rhoInput = InputData.parameterInputFactory("rho", contentType=InputTypes.FloatType)
+    samplerInitInput.addSub(rhoInput)
+    targetDistribution = InputData.parameterInputFactory("distribution")
+    samplerInitInput.addSub(targetDistribution)
+    samplingTypeInput = InputData.parameterInputFactory("samplingType", contentType=InputTypes.StringType)
+    samplerInitInput.addSub(samplingTypeInput)
+    inputSpecification.addSub(samplerInitInput)
+
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Default Constructor that will initialize member variables with reasonable
+      defaults or empty lists/dictionaries where applicable.
+      @ In, None
+      @ Out, None
+    """
+    ForwardSampler.__init__(self)
+    self.printTag = 'SAMPLER UMBRELLA'
+    self.samplingType = None
+    self.limit = None
+    self.modeLocation = 0
+    self.rho = 0
+    self.dimension = 0
+
+  def localInputAndChecks(self,xmlNode, paramInput):
+    """
+      Class specific xml inputs will be read here and checked for validity.
+      @ In, xmlNode, xml.etree.ElementTree.Element, The xml element node that will be checked against the available options specific to this Sampler.
+      @ In, paramInput, InputData.ParameterInput, the parsed parameters
+      @ Out, None
+    """
+    ForwardSampler.readSamplerInit(self,xmlNode)
+    self.modeLocation = paramInput.findFirst('samplerInit').findFirst('modeLocation').value
+    self.tailProb = paramInput.findFirst('samplerInit').findFirst('tailProbability').value
+    self.rho = paramInput.findFirst('samplerInit').findFirst('rho').value
+    self.dimension = paramInput.findFirst('samplerInit').findFirst('dimensions').value
+    self.weightTarget = paramInput.findFirst('samplerInit').findFirst('weightTarget').value
+
+    if paramInput.findFirst('samplerInit') != None:
+      if self.limit is None:
+        self.raiseAnError(IOError,self,'Umbrella sampler '+self.name+' needs the limit block (number of samples) in the samplerInit block')
+      if paramInput.findFirst('samplerInit').findFirst('samplingType')!= None:
+        self.samplingType = paramInput.findFirst('samplerInit').findFirst('samplingType').value
+        if self.samplingType not in ['uniform']:
+          self.raiseAnError(IOError,self,'Umbrella sampler '+self.name+': specified type of samplingType is not recognized. Allowed type is: uniform')
+      else:
+        self.samplingType = None
+    else:
+      self.raiseAnError(IOError,self,'Umbrella sampler '+self.name+' needs the samplerInit block')
+
+  def getStd(self, k, sigma):
+    """
+         Method to get alpha and beta values of gamma distribution based on mode location, standard deviation and tail probability
+         @ In, k, mode location of the gamma distribution specified by the user
+         @ In, tail, tail probability of the distribution
+         @ Out, sigma, standard distribution of the gamma distribution
```

Comment:
The docstrings don't match the method arguments here; this is true for many docstrings added in this PR; I'll just mention it here.

Reply:
Also, the format for the in/out docstrings are missing the "type". The format is
```python
"""
  @ In, varName, varType, description
"""
```
such as
```python
"""
  @ In, age, int, age of the unit used in this calculation
"""
```